pyquil/quantum_processor/transformers/qcs_isa_to_compiler_isa.py

```diff
@@ -167,39 +159,46 @@ def _make_rx_gates(node_id: int, characteristics: List[Characteristic]) -> List[
     return gates
 
 
-def _make_rz_gates(node_id: int) -> List[GateInfo]:
+def _make_rz_gates(node_id: int, benchmarks: List[Operation]) -> List[GateInfo]:
+    fidelity = _get_frb_sim_1q(node_id, benchmarks)
     return [
         GateInfo(
             operator=Supported1QGate.RZ,
             parameters=["_"],
             arguments=[node_id],
-            fidelity=PERFECT_FIDELITY,
+            fidelity=fidelity,
             duration=PERFECT_DURATION,
         )
     ]
 
 
+def _get_frb_sim_1q(node_id: int, benchmarks: List[Operation]) -> float:
+    frb_sim_1q = next(
```

Comment:
This will give a cryptic error message if there is no such benchmark score. `next` should be used with a default/fallback value of `None` and then you can either allow `None` as the return value or provide an error message with an assertion.